### PR TITLE
7272 Remove STEM Scholarship question from 22-1995 workflow and submission

### PR DESCRIPTION
**Pull request by elder-joseph**
## Description
As a veteran looking to update my benefits, I should not have to answer unnecessary questions about the STEM scholarship application in order to update my benefits.

[Originating Issue](https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/7272)

## Screenshots
N/A

## Acceptance criteria
- [x] STEM Scholarship Application questions are not included in the 22-1995 workflow.
      - The STEM questions
      - The readonly view on the "Review Application page"
- [x] The 22-1995 spool file does not include STEM Scholarship Application submissions.
- [x] The YTD Report does not include 22-1995 STEM information (Discuss)

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

File: src/applications/edu-benefits/10203/containers/StemEligibilityView.jsx [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+      'vads-u-color--gray-medium': !indication,
+    });
+
+  iconText = indication =>
+    indication ? `You answered yes to` : `You answered no to`;
+
+  renderCheck = (classes, title, text) => (
+    <li className="vads-u-margin-bottom--0">
+      <span className="fa-li">
+        <i className={classes} aria-hidden="true" />
```
[the remark this answers is not in the document]
Comment (bkjohnson) [reply src/applications/edu-benefits/10203/containers/StemEligibilityView.jsx:35]: Screenshots would help us make a decision here.

File: src/applications/edu-benefits/10203/containers/StemEligibilityView.jsx [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+    <div>
+      <p className="vads-u-margin-bottom--1">
+        <span className="vads-u-font-family--serif heading-level-4">
+          Based on your responses, it appears you're not eligible.
+        </span>
+        <br />
+        <br />
+        <b>Your responses:</b>
+      </p>
+      {/* eslint-disable jsx-a11y/no-redundant-roles */}
```
[the remark this answers is not in the document]
Comment (bkjohnson) [reply src/applications/edu-benefits/10203/containers/StemEligibilityView.jsx:105]: Why is `role="list"` added to the `<ul>`?

**Review by sonntag-adam — APPROVED** [review]

File: src/applications/edu-benefits/10203/containers/StemEligibilityView.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (elder-joseph) [reply src/applications/edu-benefits/10203/containers/StemEligibilityView.jsx]: This is being added to resolve a 508 related issue.

File: src/applications/edu-benefits/10203/containers/StemEligibilityView.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (elder-joseph) [reply src/applications/edu-benefits/10203/containers/StemEligibilityView.jsx]: Hi @bkjohnson this was added as a result of a UX/UI review

**Review by zurbergram — APPROVED** [review]

**Review by zurbergram — CHANGES_REQUESTED** [review]
Why is this a removal and not putting this behind a the `edu_benefits_stem_scholarship` feature flag?

As stated in the assumptions below:
1. This work is completed behind a feature flag (#7271)
2. This work is not enabled in production until the new STEM OMB form is available in production.

If you remove this code how will it be present in production still?

**Comment by zurbergram** [comment]
This is done wrong.
Correct solution should be to update function at https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/edu-benefits/1995/helpers.jsx#L22 to check if feature flag is enabled.

File: src/applications/edu-benefits/components/EducationWizard.jsx [hunk 3]
```diff
@@ -447,7 +447,7 @@ export class EducationWizard extends React.Component {
             {newBenefit === 'no' &&
               (transferredEduBenefits === 'transferred' ||
                 transferredEduBenefits === 'own') &&
-              this.getButton(form1995)}
+              this.getButton('1995')}
```
Comment (zurbergram) [inline src/applications/edu-benefits/components/EducationWizard.jsx:450]: Undo this change, this is a place where the feature flag can continued to be used and is unrelated to this story

File: src/applications/edu-benefits/1995/helpers.jsx [hunk 4]
```diff
@@ -17,6 +19,7 @@ export const displayStemEligibility = form =>
   !isEligibleForEdithNourseRogersScholarship(form);
 
 export const determineEligibilityFor1995Stem = form =>
+  environment.isProduction() &&
   form['view:determineEligibility']['view:determineEligibility'];
 
 export const display1995StemFlow = form =>
```
Comment (zurbergram) [inline src/applications/edu-benefits/1995/helpers.jsx:25]: `display1995StemFlow` also needs an `environment.isProduction()`

File: src/applications/edu-benefits/1995/config/chapters.js [hunk 5]
```diff
@@ -27,190 +28,352 @@ const {
...
-    title: 'Military History',
-    pages: {
-      // 1995-STEM related
-      activeDuty: {
-        title: 'Active Duty',
-        path: 'active-duty',
-        depends: display1995StemFlow, // 1995-STEM related
-        uiSchema: activeDuty.uiSchema,
-        schema: activeDuty.schema,
+export const chapters = environment.isProduction()
```
Comment (zurbergram) [inline src/applications/edu-benefits/1995/config/chapters.js:31]: Undo this.

In addition to updating the functions `determineEligibilityFor1995Stem` and `display1995StemFlow` in `helper.jsx` to include an `environment.isProduction()` check you should just update the initial 1995-STEM page to have depends check for `environment.isProduction()`

The function `display1995StemFlow` is used to enable/disable all 1995-STEM pages and code after going through the initial page.

```
 // related to 1995-STEM
      stem: {
        title: 'Rogers STEM Scholarship',
        path: 'benefits/stem',
        depends: environment.isProduction(),
        uiSchema: stem.uiSchema,
        schema: stem.schema,
      },
```

**Review by zurbergram — CHANGES_REQUESTED** [review]
Please make requested changes.

File: src/applications/edu-benefits/components/EducationWizard.jsx (hunk 3 above)
Comment (elder-joseph) [reply src/applications/edu-benefits/components/EducationWizard.jsx:450]: This was a part of Dustin's 10203 change, should it be left off for now?

File: src/applications/edu-benefits/1995/config/chapters.js (hunk 5 above)
Comment (elder-joseph) [reply src/applications/edu-benefits/1995/config/chapters.js:31]: I see, thanks for pointing that out. Will make the change here.

File: src/applications/edu-benefits/components/EducationWizard.jsx (hunk 3 above)
Comment (zurbergram) [reply src/applications/edu-benefits/components/EducationWizard.jsx:450]: Found the conversation about including this change, this is ok to leave in

**Review by zurbergram — APPROVED** [review]
LGTM

**Comment by elder-joseph** [comment]
@department-of-veterans-affairs/frontend-review-group Could I get a code review on this at your earliest convenience.

**Review by bkjohnson — COMMENTED** [review]
Have you considered using feature toggles instead of `environment.isProduction()`?

https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/platform/tools/feature-toggles#add-feature

**Comment by elder-joseph** [comment]
> Have you considered using feature toggles instead of `environment.isProduction()`?
> 
> https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/platform/tools/feature-toggles#add-feature

Hi @bkjohnson, I originally explored a solution using feature flags, but found that for this particular situation it would not be possible to implement without a custom container wired to redux.

**Comment by elder-joseph** [comment]
@bkjohnson If you have a recommendation for implementing a feature flag in this situation that would be great!

**Review by bkjohnson — APPROVED** [review]
I see what you mean.

**Comment by elder-joseph** [comment]
@bkjohnson not sure how this would come about, but in the future it would be nice to be able to leverage the feature flags for situations like this in the form system.

**Comment by bkjohnson** [comment]
I agree. maybe this is something @Clark-W-Griswold and I could think about.
